src/main/java/roomescape/reservation/service/ReservationService.java

```diff
@@ -68,29 +73,45 @@ public List<ReservationResponse> getAll() {
                 .toList();
     }
 
-    public List<ReservationResponse> findByFilter(final Long memberId, final Long themeId,
-                                                  final LocalDate dateFrom, final LocalDate dateTo) {
-        return reservationRepository.findByThemeIdAndMemberIdAndDateBetween(themeId, memberId, dateFrom, dateTo)
-                .stream()
+    public List<ReservationResponse> findByFilter(final Long memberId,
+                                                  final Long themeId,
+                                                  final String status,
+                                                  final LocalDate dateFrom,
+                                                  final LocalDate dateTo
+    ) {
+        return reservationRepository.findByThemeIdAndMemberIdAndStatusAndDateBetween(
+                        themeId, memberId, Status.from(status), dateFrom, dateTo
+                ).stream()
                 .map(ReservationResponse::from)
                 .toList();
     }
 
     public List<MemberReservationResponse> findByMemberId(final long memberId) {
-        return reservationRepository.findByMemberId(memberId)
-                .stream()
-                .map(MemberReservationResponse::from)
-                .toList();
+        List<Reservation> reservations = reservationRepository.findByMemberId(memberId);
+        return reservations.stream()
+                .map(reservation -> MemberReservationResponse.of(
+                        reservation,
+                        reservationRepository.countByDateAndTimeIdAndThemeIdAndCreatedAtBefore(
+                                reservation.getDate(),
+                                reservation.getTime().getId(),
+                                reservation.getTheme().getId(),
+                                reservation.getCreatedAt()
+                        )
+                )).toList();
     }
 
-    public ReservationDeleteResponse delete(final long id) {
-        validateNotExitsReservationById(id);
-        return new ReservationDeleteResponse(reservationRepository.deleteById(id));
+    public void delete(final long id) {
+        Reservation reservation = getById(id);
+        reservationRepository.delete(reservation);
+
+        Optional<Reservation> reservationWithStatusWaiting = reservationRepository.findFirstByDateAndTimeIdAndThemeIdOrderByCreatedAt(
+                reservation.getDate(), reservation.getTime().getId(), reservation.getTheme().getId()
+        );
+        reservationWithStatusWaiting.ifPresent(value -> value.setStatus(Status.RESERVATION));
```

Comment:
```suggestion
        reservationRepository.findFirstByDateAndTimeIdAndThemeIdOrderByCreatedAt(
                reservation.getDate(), reservation.getTime().getId(), reservation.getTheme().getId()
        ).ifPresent(value -> value.setStatus(Status.RESERVATION));
```
사실 이정도면 그냥 Optional을 따로 안빼도 되지 않을까싶어요.